File: Documentation/6.0/Samples/python/ClientApi/Session/Querying/HowToGetQueryStatistics.py
from typing import Callable, TypeVar

from ravendb import DocumentQuery, QueryStatistics
from ravendb.infrastructure.orders import Employee

from examples_base import ExampleBase

_T = TypeVar("_T")


class GetQueryStatistics(ExampleBase):
    def setUp(self):
        super().setUp()

    class Foo:
        # region stats_1
        def statistics(self, stats_callback: Callable[[QueryStatistics], None]) -> DocumentQuery[_T]: ...

    def test_get_query_statistics(self):
        with self.embedded_server.get_document_store("QueryStatistics") as store:
            with store.open_session() as session:
                # region stats_2
                def __statistics_callback(statistics: QueryStatistics) -> None:
                    # Read and interact with QueryStatistics here
                    total_results = statistics.total_results
                    duration_milliseconds = statistics.duration_in_ms
                    ...

                employees = list(
                    session.query(object_type=Employee)
                    .where_equals("first_name", "Robert")
                    .statistics(__statistics_callback)
                )
                # endregion

File: Documentation/6.0/Samples/php/ClientApi/Session/Querying/HowToGetQueryStatistics.php
<?php

use RavenDB\Documents\Session\DocumentQueryInterface;
use RavenDB\Samples\Infrastructure\DocumentStoreHolder;
use RavenDB\Samples\Infrastructure\Orders\Employee;

interface FooInterface {
    # region stats_1
    public function statistics(QueryStatistics &$stats): DocumentQueryInterface;
    # endregion
}

# region stats_2
class QueryStatistics
{
    private bool $isStale = false;
    private int $durationInMs = 0;
    private int $totalResults = 0;
    private int $longTotalResults = 0;
    private int $skippedResults = 0;
    private ?int $scannedResults = null;
    private ?DateTimeInterface $timestamp = null;
    private ?string $indexName = null;
    private ?DateTimeInterface $indexTimestamp = null;
    private ?DateTimeInterface $lastQueryTime = null;
    private ?int $resultEtag = null;
    private ?string $nodeTag = null;
}
# endregion

class HowToGetQueryStatistics
{
    public function samples(): void
    {
        $store = DocumentStoreHolder::getStore();
        try {

            $session = $store->openSession();
            try {
                # region stats_3
                $stats = new QueryStatistics();

                $employees = $session->query(Employee::class)
                    ->whereEquals("FirstName", "Robert")
                    ->statistics($stats)
                    ->toList();

                $totalResults = $stats->getTotalResults();
                $durationInMs = $stats->getDurationInMs();
                # endregion

            } finally {
                $session->close();
            }
        } finally {
            $store->close();
        }
    }
}
